metadata_backend/helpers/metax_api_handler.py

```diff
@@ -0,0 +1,261 @@
+                )
+                return metax_data["identifier"]
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
+
+    async def update_draft_dataset(self, collection: str, data: Dict) -> str:
+        """Update draft dataset to Metax.
+
+        Construct Metax draft dataset data from submitters' Study or Dataset and
+        send it to Metax Dataset API for update.
+
+        :param collection: schema of incomming submitters metadata
+        :param data: validated Study or Dataset data dict
+        :raises: HTTPError depending on returned error from Metax
+        :returns: Metax ID for dataset returned by Metax API
+        """
+        metax_dataset = self.minimal_dataset_template
+        # TODO: should this be changed if person updating data is different from data creator?
+        metax_dataset["metadata_provider_user"] = await self.get_metadata_provider_user()
+        if collection == "dataset":
+            dataset_data = await self.create_metax_dataset_data_from_dataset(data)
+        else:
+            dataset_data = await self.create_metax_dataset_data_from_study(data)
+        metax_dataset["research_dataset"] = dataset_data
+        LOG.info(f"Sending updated {collection} object data to Metax service.")
+
+        async with aiohttp.ClientSession() as sess:
+            resp = await sess.put(
+                f'{self.metax_url}{self.rest_route}/{data["metaxIdentifier"]}',
+                params="draft",
+                json=metax_dataset,
+                auth=aiohttp.BasicAuth(self.username, self.password),
+            )
+            status = resp.status
+            if status == 200:
+                metax_data = await resp.json()
+                LOG.info(f"Updated Metax draft dataset with ID {metax_data['identifier']} with data: {metax_dataset}")
+                return metax_data["identifier"]
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
+
+    async def delete_draft_dataset(self, metax_id: str) -> None:
+        """Delete draft dataset from Metax service.
+
+        :param metax_id: Identification string pointing to Metax dataset to be deleted
+        """
+        async with aiohttp.ClientSession() as sess:
+            resp = await sess.delete(
+                f"{self.metax_url}{self.rest_route}/{metax_id}",
+                auth=aiohttp.BasicAuth(self.username, self.password),
+            )
+            status = resp.status
+            if status == 204:
+                LOG.info(f"Deleted draft dataset {metax_id} from Metax service")
+            else:
+                # TODO: how front end should react on this??
+                reason = await resp.text()
+                raise self.process_error(status, reason)
```

Comment:
@hannyle @saulipurhonen

Reply:
this is already handled in the middleware with the redirect to the error pages, not sure why we need to do more here than raising the error. See: https://github.com/CSCfi/metadata-submitter/blob/develop/metadata_backend/api/middlewares.py#L18-L71

Reply:
:thinking: on second thought this might be a good point, we might want to display this information differently on front-end instead of redirecting to a page.

the same goes for the rest of the similar comments, we need to create an issue in front-end about it.